diff --git a/smtp.mail.inc b/smtp.mail.inc
index b220eeb..3b06150 100644
--- a/smtp.mail.inc
+++ b/smtp.mail.inc
@@ -505,7 +505,8 @@ class SmtpMailSystem implements MailSystemInterface {
 
 
     // Set other connection settings.
-    $mailer->Host = variable_get('smtp_host', '') . ';' . variable_get('smtp_hostbackup', '');
+    $mailer->Hostname = variable_get('smtp_host', '');
+    $mailer->Host = $mailer->Hostname . ';' . variable_get('smtp_hostbackup', '');
     $mailer->Port = variable_get('smtp_port', '25');
     $mailer->Mailer = 'smtp';
 
